docs/internals/protocol/dataformats.rst

```diff
@@ -0,0 +1,374 @@
+
+std::int64
+==========
+
+The :eql:type:`std::int64` values are represented as eight bytes, most
+significant byte first.
+
+For example, the ``int64`` value ``123456789987654321`` is represented as:
+
+.. code-block::
+
+    0x01 0xb6 0x9b 0x4b 0xe0 0x52 0xfa 0xb1
+
+
+.. _ref_protocol_fmt_float32:
+
+std::float32
+============
+
+The :eql:type:`std::float32` values are represented as a IEEE 754-2008 binary
+32-bit value, most significant byte first.
+
+For example, the ``float32`` value ``-15.625`` is represented as:
+
+.. code-block::
+
+    0xc1 0x7a 0x00 0x00
+
+
+.. _ref_protocol_fmt_float64:
+
+std::float64
+============
+
+The :eql:type:`std::float32` values are represented as a IEEE 754-2008 binary
+64-bit value, most significant byte first.
+
+For example, the ``float64`` value ``-15.625`` is represented as:
+
+.. code-block::
+
+    0xc0 0x2f 0x40 0x00 0x00 0x00 0x00 0x00
+
+
+.. _ref_protocol_fmt_decimal:
+
+std::decimal
+============
+
+The :eql:type:`std::decimal` values are represented as the following structure:
+
+.. code-block:: c
+
+    struct Decimal {
+        // Number of digits in digits[], can be 0.
+        uint16               ndigits;
+
+        // Weight of first digit.
+        int16                weight;
+
+        // Sign of the value
+        uint16<DecimalSign>  sign;
+
+        // Value display scale.
+        uint16               dscale;
+
+        // base-10000 digits.
+        int16                digits[ndigits];
+    };
+
+    enum DecimalSign {
+        // Positive value.
+        POS     = 0x0000;
+        // Negative value.
+        NEG     = 0x4000;
+        // NaN value.
+        NAN     = 0xC000;
+    };
+
+The decimal values are represented as a sequence of base-10000 *digits*.
+The first digit is assumed to be multiplied by *weight* * 10000, i.e. there
+are weight + 1 digits before the decimal point.  It is possible to have
+negative weight.
+
+*dscale*, or display scale, is the nominal precision expressed as number of
+base-10 digits after the decimal point.  It is always non-negative.  dscale may
+be more than the number of physically present fractional digits, implying
+significant trailing zeroes.  It is never less that the number of digits.
+
+For example, the decimal value ``-15000.6250000`` is represented as:
+
+.. code-block:: c
+
+    // ndigits
+    0x00 0x04
+
+    // weight
+    0x00 0x01
+
+    // sign
+    0x40 0x00
+
+    // dscale
+    0x00 0x07
+
+    // digits
+    0x00 0x01 0x13 0x88 0x18 0x6a 0x00 0x00
+
+
+.. _ref_protocol_fmt_bool:
+
+std::bool
+=========
+
+The :eql:type:`std::bool` values are represented as a single byte value,
+``0x01`` for ``true`` and ``0x00`` for ``false``.
+
+
+.. _ref_protocol_fmt_datetime:
+
+std::datetime
+=============
+
+The :eql:type:`std::datetime` values are represented as a 64-bit integer,
+most sigificant byte first.  The value is the number of *microseconds*
+between the encoded datetime and January 1st 2000, 00:00 UTC.
```

Comment:
Maybe it's worth mentioning that `946684800` is the *proper* number of seconds that should be used to convert between 2000/1970 timestamps.